domain/src/validation.js

```diff
@@ -40,21 +40,46 @@ function checkItem(item) {
         });
     }
     let isValid = !errors.length;
-    if (isValid) {
-        if (item.critChance === 0)
-            delete item.critChance;
-        if (item.critDmg === 0)
-            delete item.critDmg;
-    } else
+    if (!isValid)
         item = undefined;
-
     return {
         item,
         isValid,
         errors
     };
 }
 
+// > checkChar({atkSpd:'asd', dmg: 3, critChance: 5, critDmg: -6})
+// {char: undefined, isValid: false, errors: [{id: "mustBeNumber", properties: ["atkSpd"]}, {id: "mustNotBeNegative", properties: ['critDmg']}]}
+function checkChar(char) {
+    char = _.pick(char, ['_id', 'atkSpd', 'dmg', 'critChance', 'critDmg']);
+    let errors = [];
+    let notNumbers = filterNotNumbers(char, ['atkSpd', 'dmg', 'critChance', 'critDmg']);
+    if (notNumbers.length) {
+        errors.push({
+            id: "mustBeNumber",
+            properties: notNumbers
+        });
+    }
+    let negative = filterNegative(char, ['atkSpd', 'dmg', 'critChance', 'critDmg'], notNumbers);
```

Comment:
It is completely valid for a char to have negative stat modifiers, remove this filter. And account for that in calculating dps.